pkg/users/accounts-list.js

```diff
@@ -18,72 +18,277 @@
 import { account_create_dialog } from "./account-create-dialog.js";
+import { delete_account_dialog } from "./delete-account-dialog.js";
+import { logoutAccountDialog } from "./logout-account-dialog.js";
+import { lockAccountDialog } from "./lock-account-dialog.js";
+import { ListingTable } from 'cockpit-components-table.jsx';
+import { EmptyStatePanel } from 'cockpit-components-empty-state.jsx';
+import * as timeformat from "timeformat.js";
 
 const _ = cockpit.gettext;
 
-function AccountItem({ account, current }) {
-    function click(ev) {
-        if (!ev)
-            return;
+const UserActions = ({ account }) => {
+    const [isKebabOpen, setKebabOpen] = useState(false);
 
-        if (ev.type === 'click' && ev.button !== 0)
-            return;
+    const actions = [
+        <DropdownItem key="edit-user"
+                      onClick={ev => { ev.preventDefault(); cockpit.location.go(account.name) }}>
+            {_("Edit user")}
+        </DropdownItem>,
+    ];
 
-        if (ev.type === 'keypress' && ev.key !== "Enter")
-            return;
+    superuser.allowed && actions.push(
+        <DropdownSeparator key="separator-0" />,
+        <DropdownItem key="log-user-out"
+                      isDisabled={account.uid === 0 || !account.loggedIn}
+                      onClick={() => logoutAccountDialog(account)}>
+            {_("Log user out")}
+        </DropdownItem>,
+        <DropdownSeparator key="separator-1" />,
+        <DropdownItem key="lock-account"
+                      isDisabled={account.isLocked}
+                      onClick={() => lockAccountDialog(account)}>
+            {_("Lock account")}
+        </DropdownItem>,
+        <DropdownItem key="delete-account"
+                      isDisabled={account.uid === 0}
+                      className={account.uid === 0 ? "" : "delete-account-red"}
+                      onClick={() => delete_account_dialog(account)}>
+            {_("Delete account")}
+        </DropdownItem>,
+    );
+
+    const kebab = (
+        <Dropdown toggle={<KebabToggle onToggle={isOpen => setKebabOpen(isOpen)} />}
+                isPlain
+                isOpen={isKebabOpen}
+                position="right"
+                dropdownItems={actions} />
+    );
+    return kebab;
+};
+
+const getAccountRow = (account, current, groups) => {
+    const admins = ['sudo', 'root', 'wheel'];
+    const userGroups = [];
+
+    groups.forEach((group) => {
+        if (group.userlist.find(accountName => accountName === account.name)) {
+            const isAdmin = !!admins.find(adm => adm === group.name);
+            userGroups.push({ name: group.name, members: group.userlist.length, isAdmin: isAdmin });
+        }
+    });
+
+    userGroups.sort((a, b) => {
+        if (a.isAdmin)
+            return -1;
+        if (b.isAdmin)
+            return 1;
+        if (a.members === b.members)
+            return a.name.localeCompare(b.name);
+        else
+            return b.members - a.members;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17830-20221116-150219-482f17ff-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)